bindings/core/src/method_handler/wallet.rs

```diff
@@ -210,8 +210,18 @@ pub(crate) async fn call_wallet_method_internal(wallet: &Wallet, method: WalletM
             let implicit_account_creation_address = wallet.implicit_account_creation_address().await?;
             Response::Bech32Address(implicit_account_creation_address)
         }
-        WalletMethod::PrepareImplicitAccountTransition { output_id } => {
-            let data = wallet.prepare_implicit_account_transition(&output_id).await?;
+        WalletMethod::PrepareImplicitAccountTransition {
+            output_id,
+            public_key,
+            bip_path,
+        } => {
+            let data = if public_key.is_some() {
+                wallet
+                    .prepare_implicit_account_transition(&output_id, public_key)
+                    .await?
+            } else {
+                wallet.prepare_implicit_account_transition(&output_id, bip_path).await?
```

Comment:
Is there nothing we can do with an Option method to do one or the other without the if ?

Reply:
sad reacts only